non-normative/recipes.adoc

```diff
@@ -0,0 +1,32 @@
+[[recipe-brs-i-guidance]]
+===== BRS-I Recipe Guidance
+
+Systems compliant to BRS-I can successfully boot an existing generic
+operating system image without system-specific customizations, yet
+this might result in an unoptimized experience and non-functioning
+I/O devices until further software updates are activated.
+
+The best analogy would be a typical Intel Architecture motherboard from
+the early 2000s: you could install an OS on it, but the built-in graphics
+might be low-resolution and the sound, built-in network port or power
+management might not work out of the box. You could subsequently load
+the right drivers from the media coming with the board or fetch newest
+ones using a well-supported network adapter.
+
+Heterogeneous performance harts (e.g. mix of "performance" and "efficiency"
+harts) is a great example of a feature outside the current scope of BRS-I,
+yet with potential for adverse effects to a generic operating system
+written against the BRS-I spec. Such a feature does not have a standard
+way of being detected by an OS, leading to scheduling anomalies in
+operating systems that are not specially adapted. Consider two identical
+threads that, unbeknownst to the OS, get scheduled on harts with different
+performance characteristics. This might see dramatic difference in forward
+progress being made, with unexpected delays, timeouts or even crashes
+posible. Worse, there could be little to no system messages pointing to
+the cause of such behavior. Thus, such a feature would need to be made
+*opt-in* for BRS-I compliance. This could be done implicitly via loading
```

Comment:
Not sure what exactly do you mean by *opt-in*. BRS-I compliance shouldn't include those features at all. Correct ?

Reply:
If your system has a feature that prevents BRS-I operating systems from operating correctly (crashes, hangs, timeouts, etc) unless a custom kernel is installed (that is aware of your system's non-standard feature or of your non-standard way of exposing a feature), then this feature cannot be automatically enabled at boot. That's what this is saying. The big.LITTLE example is a good example of what I mean.

Reply:
Got it. Any custom feature prevents the platform to boot BRS-I operating systems makes it incompatible.
However, if the platform boots while disabling that feature but enables at runtime, it is called *opt-in* for the sake of compliance.